plumbing/format/packfile/parser.go

```diff
@@ -35,6 +35,33 @@ type Observer interface {
 	OnFooter(h plumbing.Hash) error
 }
 
+type StatusObserver struct {
+	statusChan plumbing.StatusChan
+	update     plumbing.StatusUpdate
+}
+
+func NewStatusObserver(sc plumbing.StatusChan) *StatusObserver {
+	return &StatusObserver{statusChan: sc}
+}
+func (so *StatusObserver) OnHeader(count uint32) error {
+	so.update.Stage = plumbing.StatusFetch
+	so.update.ObjectsTotal = int(count)
+	so.statusChan.SendUpdate(so.update)
+
+	return nil
+}
+func (so *StatusObserver) OnInflatedObjectHeader(t plumbing.ObjectType, objSize int64, pos int64) error {
+	so.statusChan.SendUpdate(so.update)
```

Comment:
Shouldn't these lines be reversed?  Seems like it would never send an update when `ObjectDone == ObjectsTotal`.  It's been a while since I thought about this though, so I might be forgetting something important.

Reply:
No, since this is called before the object is done. 

However we can send the ObjectsDone == ObjectsDone status update on footer. Added that.